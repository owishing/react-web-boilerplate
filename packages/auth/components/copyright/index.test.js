import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Copyright } from '.';

function shallowRenderWithProps(props = {}) {
  return shallow(<Copyright {...props} />);
}

test('Render right Copyright when no year prop provided.', () => {
  const component = shallowRenderWithProps();
  expect(toJson(component)).toMatchSnapshot();
});

test('Render right Copyright when right year prop provided.', () => {
  const component = shallowRenderWithProps({ year: 2021 });
  expect(toJson(component)).toMatchSnapshot();
});
