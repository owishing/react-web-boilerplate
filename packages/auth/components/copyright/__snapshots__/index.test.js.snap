// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render right Copyright when no year prop provided. 1`] = `
<WithStyles(ForwardRef(Typography))
  align="center"
  color="textSecondary"
  variant="body2"
>
  Copyright © 
  <WithStyles(ForwardRef(Link))
    color="inherit"
    href="https://www.theknot.com/"
  >
    The Knot
  </WithStyles(ForwardRef(Link))>
   
  2020
  .
</WithStyles(ForwardRef(Typography))>
`;

exports[`Render right Copyright when right year prop provided. 1`] = `
<WithStyles(ForwardRef(Typography))
  align="center"
  color="textSecondary"
  variant="body2"
>
  Copyright © 
  <WithStyles(ForwardRef(Link))
    color="inherit"
    href="https://www.theknot.com/"
  >
    The Knot
  </WithStyles(ForwardRef(Link))>
   
  2021
  .
</WithStyles(ForwardRef(Typography))>
`;
